### PR TITLE
Example fixes

### PR DESCRIPTION
**Pull request by einarf**
### Description

Misc fixes when going through the examples directory.

### List of changes

- Ensure example windows support 4k and retina displays (viewport was incorrect)
- GLFW example did not specify core forward compat context causing issues for some setups
- Simplify README: Only require `pip install -r requirements.txt`
- Added matplotlib to requirements.txt
- A couple of `.gitignore` entries

### CODE REVIEW

File: examples/04_colors_and_texture.py [hunk 1]
```diff
@@ -6,7 +6,7 @@
 
 import moderngl
 import numpy as np
-from objloader import Obj
+from moderngl_obj import Obj
```
Comment (cprogrammer1994) [inline examples/04_colors_and_texture.py]: `moderngl_obj` is deprecated and `objloader` should be used instead.

Comment (einarf) [reply examples/04_colors_and_texture.py]: I reverted that further down when I checked the age of the packages on pipy

**Comment by einarf** [comment]
Ignore the second commit 😄

**Comment by einarf** [comment]
I can submit a cleaner pr without the second commit if you want, or do you auto-squash on merge? This was just things I picked up in my journey through the examples directory over time, so I derped on the obj package initially.

**Comment by cprogrammer1994** [comment]
Oh, I just realized that you already reverted that change. The final diff looks great. merging

**Comment by cprogrammer1994** [comment]
Thank you!

**Pull request merged**